src/sharing/components/ShareAutocomplete.jsx

```diff
@@ -6,6 +6,26 @@ import Recipient from './Recipient'
 
 import { getPrimaryEmail } from '..'
 
+// TODO: sadly we have different versions of contacts' doctype to handle...
+// A migration tool on the stack side is needed here
+const emailMatch = (input, contact) => {
+  if (!contact.email) return false
+  const regex = new RegExp(input)
+  if (Array.isArray(contact.email)) {
+    return contact.email.some(email => regex.test(email.address))
+  }
+  return regex.test(contact.email)
+}
+
+const cozyUrlMatch = (input, contact) => {
+  if (!contact.cozy || !contact.url) return false
+  const regex = new RegExp(input)
+  if (contact.cozy && Array.isArray(contact.cozy)) {
+    return contact.cozy.some(cozy => regex.test(cozy.url))
+  }
+  return regex.test(contact.url)
+}
```

Comment:
I see a similar pattern than `const emailMatch` here. Can't we refactor something? With HOF for instance.

Reply:
There's no 3rd copy ;) Yes, there is a pattern, but I found the possible refactoring inelegant because you have to pass 2 property names as arguments, like:
```js
const isMatching = (input, contact, propertyLevel1, propertyLevel2) => { ... }
```

Reply:
While commenting your code, I did some tries not very beautiful.

We will find some elegant way as soon as we will take the time to get some fresh air for our brain. We need to take a walk in mountain or in forest.